packages/react-storage/src/components/StorageBrowser/controls/hooks/usePagination.ts

```diff
@@ -0,0 +1,31 @@
+import { PaginationProps } from '../../composables/Pagination';
+import { useControlsContext } from '../context';
+
+export type UsePagination = () => {
+  props?: PaginationProps;
+};
+
+export const usePagination: UsePagination = () => {
+  const { data } = useControlsContext();
+  if (!data?.pagination) {
+    return {};
+  }
+
+  const {
+    currentPage,
+    disableNext,
+    disablePrevious,
+    handlePaginateNext,
+    handlePaginatePrevious,
+  } = data.pagination;
+
+  return {
+    props: {
```

Comment:
Sorry for the churn here but we are going to flatten this out to return props from the hook directly.

Also, since there's nothing being altered here, we can just feed the props straight through, I think e.g.
```
return data.pagination
```

Reply:
Yep, this is the better way. Updated